src/workerd/api/crypto-impl-asymmetric.c++

```diff
@@ -644,12 +711,36 @@ kj::Maybe<T> fromBignum(kj::ArrayPtr<kj::byte> value) {
   return asUnsigned;
 }
 
-void validateRsaParams(int modulusLength, kj::ArrayPtr<kj::byte> publicExponent) {
+void validateRsaParams(int modulusLength, kj::ArrayPtr<kj::byte> publicExponent,
+    bool warnImport = false) {
   // The W3C standard itself doesn't describe any parameter validation but the conformance tests
   // do test "bad" exponents, likely because everyone uses OpenSSL that suffers from poor behavior
   // with these bad exponents (e.g. if an exponent < 3 or 65535 generates an infinite loop, a
   // library might be expected to handle such cases on its own, no?).
 
+  // TODO(soon): We should also enforce these limitations on imported keys. To see if this breaks
+  // existing scripts only provide a warning in sentry for now.
+  if (warnImport) {
+    if (modulusLength % 8 || modulusLength < 256 || modulusLength > 16384) {
+      static bool logOnce KJ_UNUSED = ([modulusLength] {
+        KJ_LOG(WARNING, "Imported RSA key has invalid modulus length ", modulusLength, ".");
+        return true;
+      })();
+    }
+    KJ_IF_MAYBE(v, fromBignum<unsigned>(publicExponent)) {
+      if (*v != 3 && *v != 65537) {
+        static bool logOnce KJ_UNUSED = ([v] {
+          KJ_LOG(WARNING, "Imported RSA key has invalid publicExponent ", *v,".");
+          return true;
+        })();
```

Comment:
No action for this PR but just noting... this is a common enough pattern now that we could probably justify adding a new `LOG_WARNING_ONCE("...")` macro to simplify.

Reply:
That's a good suggestion! Looks like this approach is not used that much elsewhere, but if I use it again in the future I'll try adding a macro